hist/histv7/test/hist_engine.cxx

```diff
@@ -158,3 +158,78 @@ TEST(RHistEngine, FillInvalidNumberOfArguments)
    EXPECT_NO_THROW(engine2.Fill(1, 2));
    EXPECT_THROW(engine2.Fill(1, 2, 3), std::invalid_argument);
 }
+
+TEST(RHistEngine, FillWeight)
+{
+   static constexpr std::size_t Bins = 20;
+   const RRegularAxis axis(Bins, 0, Bins);
+   RHistEngine<float> engine({axis});
+
+   engine.Fill(-100, RWeight(0.25));
+   for (std::size_t i = 0; i < Bins; i++) {
+      engine.Fill(i, RWeight(0.1 + i * 0.03));
+   }
+   engine.Fill(100, RWeight(0.75));
```

Comment:
Would it make sense to use at least two dimensions in the test?
Errors stemming from incorrectly computing the global index won't show in this test (but might well be covered by other tests, hence my question).